user/find.c

```diff
@@ -0,0 +1,84 @@
+#include "kernel/types.h"
+#include "kernel/stat.h"
+#include "user/user.h"
+#include "kernel/fs.h"
+
+#define MAX_PATH_SIZE 100
+
+char *pathbase(char *path) {
+    char *p;
+    static char buf[MAX_PATH_SIZE + 1];
+
+    // Find the first character after the last slash.
+    for (p = path + strlen(path); p >= path && *p != '/'; --p)
+        ;
+
+    return strcpy(buf, p + 1);
+}
+
+char *pathjoin(char *path, char *prefix, char *dst) {
```

Comment:
```suggestion
static char *pathjoin(char *dst, const char *path, const char *prefix, ) {
```

Usually the copy destination dst is the first parameter, this is a convention. Declare untouched arrays "const" to let compilers do more assumptions.

Reply:
Note that `const char*` means we do not store bytes to the pointer points to, the pointer it self may change. `char * const` means the pointer address should not change, always points to the same object, but its contents may change. You should disambiguate these types:

```
const char *
char * const
const char * const
```